plenum/common/stashing_router.py

```diff
@@ -93,35 +93,29 @@ def pop_all(self) -> Iterable[Tuple]:
 class StashingRouter(Router):
     Handler = Callable[..., Optional[Tuple[int, str]]]
 
-    def __init__(self, limit: int, replica_unstash: Callable=None):
+    def __init__(self, limit: int, buses: List[Router], unstash_handler: Callable = None):
         super().__init__()
         self._limit = limit
         self._logger = getlogger()
         self._queues = {}  # type: Dict[int, StashingQueue]
         # TODO: This call has been added to saving the old message order in the list.
         # This is a replica's method that moves the message to the inBox, rather than
         # calling the handler immediately, as the default router does.
-        self._replica_unstash = replica_unstash
+        self._unstash_handler = unstash_handler
         self._subscriptions = Subscription()
+        self._buses = buses
 
     def set_sorted_stasher(self, code: int, key: Callable):
         self._queues[code] = SortedStash(self._limit, key)
 
-    def subscribe(self, message_type: Type, handler: Handler, allow_override=False) -> Router.SubscriptionID:
-        if not allow_override and message_type in self.message_types:
-            raise LogicError("Trying to assign handler {} for message type {}, "
-                             "but another handler is already assigned {}".
-                             format(handler, message_type, self.handlers(message_type)))
+    def subscribe(self, message_type: Type, handler: Handler) -> Router.SubscriptionID:
+        # TODO: subscribe to one bus only
+        for bus in self._buses:
+            self._subscriptions.subscribe(bus,
+                                          message_type,
+                                          partial(self._process, handler))
```

Comment:
I guess unsubscribe needs to be implemented as well

Reply:
Agree, but out of scope of this PR, especially taking into account that we always unsubscribe from all only (unsubscribe by all the services when the replica unsubscribes)